src/PointwiseFunctions/AnalyticSolutions/NewtonianEuler/Sinusoid.hpp

```diff
@@ -0,0 +1,121 @@
+//// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <limits>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/NewtonianEuler/Sources/NoSource.hpp"
+#include "Evolution/Systems/NewtonianEuler/Tags.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/AnalyticSolution.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/IdealFluid.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace PUP {
+class er;  // IWYU pragma: keep
+}  // namespace PUP
+/// \endcond
+
+// IWYU pragma: no_include <pup.h>
+
+namespace NewtonianEuler {
+namespace Solutions {
+
+/*!
+ * \brief A sinusoidal density profile advecting through the domain.
+ *
+ * Uses an ideal fluid equation of state with \f$\gamma=1.4\f$. The solution is
+ *
+ * \f{align*}{
+ * \rho(x, y, z, t) =& 1 + 0.2 \sin(x + y + z - t) \\
+ * v^i(x, y, z, t) = v^i_{\mathrm{initial}} \\
+ * p(x, y, z, t) = 1
+ * \f}
+ *
+ * with the constraint that
+ *
+ * \f{align*}{
+ *  \sum_{i} v^i = 1
+ * \f}
```

Comment:
For my understanding, why this contraint?

Reply:
If I remember correctly it's only a solution if this is satisfied. It's been a while since I plugged it all in and did the math, but that's what I got out :)

Reply:
I agree... I didn't work the solution out fully, but just enough to see how this is needed from balancing the d(rho)/dt with the div(flux(rho)) terms 👍